addons/udes_delivery_control/models/stock_picking.py

```diff
@@ -0,0 +1,115 @@
+    "u_backload_end_date",
+    "u_backload_time_taken",
+]
+
+
+class StockPicking(models.Model):
+
+    _inherit = "stock.picking"
+
+    u_is_delivery_control = fields.Boolean(
+        related="picking_type_id.u_is_delivery_control", readonly=1
+    )
+    u_extras_id = fields.Many2one("udes_picking_extras", copy=False)
+    # Unloading Fields
+    u_supplier_id = fields.Many2one(
+        "res.partner",
+        default=lambda self: self._default_u_supplier_id(),
+        related="u_extras_id.u_supplier_id",
+    )
+    u_user_id = fields.Many2one("res.users", related="u_extras_id.u_user_id",)
+    u_delivery_note = fields.Char(related="u_extras_id.u_delivery_note")
+    u_location_id = fields.Many2one("stock.location", related="u_extras_id.u_location_id")
+    u_vehicle_scheduled_date = fields.Datetime(related="u_extras_id.u_vehicle_scheduled_date",)
+    u_vehicle_arrival_date = fields.Datetime(related="u_extras_id.u_vehicle_arrival_date",)
+    u_week = fields.Integer(related="u_extras_id.u_week")
+    u_unloading_start_date = fields.Datetime(related="u_extras_id.u_unloading_start_date")
+    u_unloading_end_date = fields.Datetime(related="u_extras_id.u_unloading_end_date")
+    # TODO Add compute method to this field, add float_time widget
+    u_time_taken = fields.Float(readonly=True, related="u_extras_id.u_time_taken")
+    u_vehicle_type = fields.Char(related="u_extras_id.u_vehicle_type")
+    u_trailer_number = fields.Char(related="u_extras_id.u_trailer_number")
+    u_lane_number = fields.Char(related="u_extras_id.u_lane_number")
+    u_pallet_count = fields.Integer(related="u_extras_id.u_pallet_count")
+    u_stillage_count = fields.Integer(related="u_extras_id.u_stillage_count")
+    u_box_count = fields.Integer(related="u_extras_id.u_box_count")
+    u_is_planned = fields.Boolean(related="u_extras_id.u_is_planned",)
+    u_is_late = fields.Boolean(related="u_extras_id.u_is_late",)
+    u_is_fast_track = fields.Boolean(related="u_extras_id.u_is_fast_track",)
+
+    # Back Loading Fields
+    u_is_backload = fields.Boolean(related="u_extras_id.u_is_backload",)
+    u_backload_supplier = fields.Many2one(
+        "res.partner",
+        default=lambda self: self._default_u_supplier_id(),
+        related="u_extras_id.u_backload_supplier",
+    )
+    u_backload_pallet_count = fields.Integer(related="u_extras_id.u_backload_pallet_count",)
+    u_backload_stillage_count = fields.Integer(related="u_extras_id.u_backload_stillage_count",)
+    u_backload_cover_count = fields.Integer(related="u_extras_id.u_backload_cover_count",)
+    u_backload_reject_count = fields.Integer(related="u_extras_id.u_backload_reject_count",)
+    u_backload_start_date = fields.Datetime(related="u_extras_id.u_backload_start_date")
+    u_backload_end_date = fields.Datetime(related="u_extras_id.u_backload_end_date")
+    u_backload_time_taken = fields.Float(related="u_extras_id.u_backload_time_taken",)
+
+    def _create_picking_extras_data(self, values):
```

Comment:
Again, as this isn't generic functionality but instead just for Delivery Control module I think the function name should reference delivery control to make it a bit clearer.

Reply:
Agreed to keep the existing functionality as it is so that in the future it can be move to separate module